src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -96,14 +96,22 @@ class InterceptedMethods extends MiniPhaseTransform {
         s"that means the intercepted methods set doesn't match the code")
       tree
     }
-    lazy val Select(qual, _) = tree.fun
+    lazy val qual = tree.fun match {
+      case Select(qual, _) => qual
+      case Ident(name) => EmptyTree
```

Comment:
this should also handle idents. Here's an example illustrating how to do it: https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/backend/jvm/DottyBackendInterface.scala#L428-L438